response.go

```diff
@@ -159,6 +160,14 @@ func getResponseContent(opChain *chain, resp *http.Response) []byte {
 	return content
 }
 
+func (r *Response) getContent(opChain *chain) []byte {
```

Comment:
getContent may report failures. Previously it was invoked in constructor, and if it reported failure, all subsequent method calls became no-op because each method has this check at the beginning:

```go
	if opChain.failed() {
		return
	}
```

Now getContent is not invoked in constructor, and failure may happen in the middle of a method when it invokes getContent. We need to handle this failures: if getContent fails, we should abort method early.

Let's add bool flag to getContent:

```go
func (r *Response) getContent(opChain *chain) ([]byte, bool)
```

And everywhere where content is needed, use it like this:

```go
func (r *Response) Body() *String {
	opChain := r.chain.enter("Body()")
	defer opChain.leave()

	if opChain.failed() {
		return newString(opChain, "")
	}

	content, ok := r.getContent(opChain)
	if !ok {
		return newString(opChain, "")
	}

	return newString(opChain, string(content))
}
```

Here we make r.getContent call early right after opChain.failed() check, and we exit from method if it fails.

Reply:
We will also need to add a test that checks how all these methods (Body, Text, Form, etc) handle getContent failures.

This test should set body to a reader that will fail, then construct response, call corresponding method (e.g. Body) and check that it returned non-nil but empty value and that the chain both in response and in returned value is marked as failed.